pkg/controllers/provisioningv2/rke2/machineprovision/args.go

```diff
@@ -85,6 +86,12 @@ func (h *handler) getArgsEnvAndStatus(typeMeta meta.Type, meta metav1.Object, da
 		Data: map[string][]byte{},
 	}
 
+	if os.Getenv("HTTP_PROXY") != "" || os.Getenv("HTTPS_PROXY") != "" {
```

Comment:
is there a reason to not copy them even if they are empty?

Reply:
Only not having blank environment variables passed to the job.

Reply:
> Only not having blank environment variables passed to the job.
Not sure what exactly do you mean by this.

It's a bit weird that we check if any of two is not empty but then setting all 3 as it leaves the possibility of 1 or 2 of them be empty. I would say we either set them regardless or set each only if that specific value is not empty.

Reply:
Updated.